src/nvme/mi-mctp.c

```diff
@@ -220,6 +226,200 @@ static bool nvme_mi_mctp_resp_is_mpr(void *buf, size_t len,
 	return true;
 }
 
+static int nvme_mi_mctp_aem_fd(struct nvme_mi_ep *ep)
+{
+	struct nvme_mi_transport_mctp *mctp;
+
+	if (ep->transport != &nvme_mi_transport_mctp) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	if (!ep->transport_data) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	mctp = ep->transport_data;
+	return mctp->sd_aem;
+}
+
+static int nvme_mi_mctp_aem_purge(struct nvme_mi_ep *ep)
+{
+	char buffer[1024];
+	ssize_t bytes_read;
+
+	struct nvme_mi_transport_mctp *mctp = ep->transport_data;
+
+	// Set the socket to non-blocking mode
```

Comment:
Should it not be in non-blocking mode already? Otherwise your aem_read() may block, which is not what you want.